librz/core/cmd/cmd_analysis.c

```diff
@@ -6153,6 +5792,366 @@ RZ_IPI RzCmdStatus rz_il_vm_status_handler(RzCore *core, int argc, const char **
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_analysis_graph_dataref_handler(RzCore *core, int argc, const char **argv) {
+	char format = argc > 1 ? argv[1][0] : ' ';
+	if (format == '*') {
+		rz_core_analysis_datarefs(core, core->offset);
+	} else {
+		rz_core_agraph_reset(core);
+		rz_core_cmdf(core, ".aga* @ %" PFMT64u ";", core->offset);
+		rz_core_agraph_print(core, -1, format);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_dataref_global_handler(RzCore *core, int argc, const char **argv) {
+	char format = argc > 1 ? argv[1][0] : ' ';
+	if (format == '*') {
+		ut64 from = rz_config_get_i(core->config, "graph.from");
+		ut64 to = rz_config_get_i(core->config, "graph.to");
+		RzListIter *it;
+		RzAnalysisFunction *fcn;
+		rz_list_foreach (core->analysis->fcns, it, fcn) {
+			if ((from == UT64_MAX && to == UT64_MAX) || RZ_BETWEEN(from, fcn->addr, to)) {
+				rz_core_analysis_datarefs(core, fcn->addr);
+			}
+		}
+	} else {
+		rz_core_agraph_reset(core);
+		rz_core_cmdf(core, ".agA*;");
+		rz_core_agraph_print(core, -1, &format);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_callgraph_function_handler(RzCore *core, int argc, const char **argv) {
+	char format = argc > 1 ? argv[1][0] : ' ';
+	switch (format) {
+	case 'v':
+	case 't':
+	case 'k':
+	case 'w':
+	case ' ': {
+		core->graph->is_callgraph = true;
+		rz_core_agraph_reset(core);
+		// TODO: Use the API here
+		rz_core_cmdf(core, ".agc* @ %" PFMT64u "; agg%s;", core->offset, &format);
+		core->graph->is_callgraph = false;
+		break;
+	}
+	case 0:
+		// FIXME: WHY IT IS DIFFERENT FROM PREVIOUS ONE?
+		core->graph->is_callgraph = true;
+		rz_core_agraph_reset(core);
+		rz_core_cmd0(core, ".agc* $$");
+		rz_core_agraph_print_ascii(core);
+		core->graph->is_callgraph = false;
+		break;
+	case 'g':
+		rz_core_analysis_callgraph(core, core->offset, RZ_GRAPH_FORMAT_GMLFCN);
```

Comment:
I detected a lot of similar patterns

```c
switch(format){
    case _
        rz_core_analysis_callgraph(..., _)
}
```

Maybe it would be better

```c
rz_core_analysis_callgraph(..., rz_core_graph_format_from_string(format))
```

Reply:
Yes, good idea. I think we should make the format a full word instead of a letter, now that it's not a part of the command anymore. Something like `agf gml` instead of `agfg` or `agf g`. @ret2libc @thestr4ng3r @wargio thoughts about this? Autocompletion should solve the problem. We could even keep one-letter parsing for now as well, but prefer the full names for the format.